### ISSUE TITLE
`<SubmitButton />` no longer automatically disables when the form is invalid

### ISSUE THREAD

**Issue by heuristicAL** [jannikbuschke/formik-antd#125]
When we use a `<SubmitButton />` inside of a `<Formik>` element, the button does not automatically disable itself when there are validation errors or when the form is not dirty.

This can be seen in the [stock formik-antd 2.0 demo](https://codesandbox.io/s/github/jannikbuschke/formik-antd-example/tree/release/v2.0) in file `sample-form.tsx`

**Comment by jannikbuschke** [comment]
Hi @heuristicAL,

I have seen some issues regarding the "smartness" of this component (at least #87, #73). At some point I tried to improve the api to allow to customize when this component is disabled by providing a `disable` callback prop that exposes formikHelpers. I.e. you then could do something like this:
```jsx
<SubmitButton disable={form => !form.isDirty && form.submitCount > 0 } />
```

Then I thought Im making this too complicated and the mantra of this library is to make everything as simple as possible. So I decided to just drop the functionality (also less stuff to document). :)

Maybe also not the best decision.
Im up for readding this, maybe its also possible to introduce some sort of `mode: "smart" | "stupid"` option.
Do you think this improves this lib overall or is better implemented?

Another idea that just popped up: We could add a `<SmartButtonContainer>` that sets the `disabled` (or even others like text and tooltip) on its children depending on some form state.

**Comment by heuristicAL** [comment]
Hey @jannikbuschke ,

I see where you're coming from, trying to make the library as simple as possible. I applaud that!
That being said, IMHO: the `Submit` and `Reset` buttons should enable and disable automatically (I believe `Reset` still works as it used to and only enables when the form has been edited). My thoughts behind this are that, if you DO want a button that is just always enabled and submits the form, it is still relatively simple to simply use the base antd `Button` control do:
`<Button click={formProps.submit} />`
I say this because I see both of these controls as "drop-in and works" options for what is usually repetitive form functionality.

Love the lib! Keep up the great work and if you point me in the right direction, I can make my own PR to re-enable this :)

**Comment by jannikbuschke** [comment]
v1.9.0-beta.5 should resolve the issue. Feel free to reopen if otherwise.

**Comment by jannikbuschke** [comment]
@all-contributors add @heuristicAL for ideas

**Comment by Slus** [comment]
@jannikbuschke Would it also be possible to add this to 2.0.0-beta.2?

Thanks!

**Comment by jannikbuschke** [comment]
Hi @suarez-agu, this should be available in 2.0.0-beta.2.
